src/components/modals/profile/library/BaseProfileLibraryFolderImportModal/ImportSection.vue

```diff
@@ -87,29 +87,18 @@ export default {
       )
     },
     formatFile (file) {
-      const reader = new jsmediatags.Reader(
-        file
-      )
-
-      const handleSuccess = ({ tags }) => {
+      musicMetadata.parseFile(file.path).then(metadata => {
```

Comment:
@Borewit Can it be done like this?
```
musicMetadata.parseFile(
  file.path
).then(
  handleSuccess
).catch(
  handleError
)
```

Reply:
I understand you suggestion, because that is more readable.
I made that mistake until someone explained me this is an anti-pattern.
Reason being, it does not do the same thing. 
The `catch` function will also catch anything thrown in `handleSuccess()` (the `then` block).

It can be written async/await style:

```js
async formatFile(file) {
  let metadata;
  try {
    metadata await = musicMetadata.parseFile(file.path);
  } catch(error) {
    // failure
    return;
  }
  // success;
}
```

Reply:
@Borewit
> The catch function will also catch anything thrown in handleSuccess() (the then block).

I'm okay with that.
Seems like there shouldn't be any errors in `handleSuccess` anyway.

Reply:
Okay, then it's fine